tools/cellpose/cellpose.xml

```diff
@@ -0,0 +1,154 @@
+    </requirements>
+    <stdio>
+        <exit_code range="1:" level="fatal" description="Error occurred. Please check Tool Standard Error"/>
+    </stdio>
+    <version_command>echo "@VERSION@"</version_command>
+    <command>
+         <![CDATA[
+        export CELLPOSE_LOCAL_MODELS_PATH='cellpose_models' &&
+        mkdir -p segmentation &&
+        ln -s '${img_in}' ./image.${img_in.ext} &&
+		
+        python '$__tool_directory__/cp_segmentation.py'
+            --inputs '$inputs'
+            --img_path ./image.${img_in.ext}
+            --img_format '${img_in.ext}'
+            --output_dir ./segmentation
+
+        ]]>
+
+    </command>
+    <configfiles>
+        <inputs name="inputs" />
+    </configfiles>
+    <inputs>
+        <param name="img_in" type="data" format="ome.tiff,tiff,jpeg,png" label="Choose the image file for segmention (usually after registration)"/>
+        <param name="model_type" type="select" label="Choose the pre-trained model type">
+            <option value="nuclei" selected="true">nuclei</option>
+            <option value="cyto">cyto</option>
+            <option value="cyto2">cyto2</option>
+            <option value="cyto3">cyto3</option>
+        </param>
+        <param argument="chan" value="0" type="integer" label="Type in the channel to segment" help="0=grayscale, 1=red, 2=green, 3=blue"/>
+        <param argument="chan2" value="0" type="integer" optional="true" label="Type in the channel for nuclei segmatation (Optional)" help="0=None (will set to zero), 1=red, 2=green, 3=blue"/>
+        <param name="chan_first" type="boolean" truevalue="booltrue" falsevalue="boolfalse" checked="false" label="Use the reshaped data with channel as the first dimension?"/>
+        <param name="show_segmentation" type="boolean" truevalue="booltrue" falsevalue="boolfalse" checked="true" label="Whether to show segmentation?"/>
+	    <param name="use_gpu" type="boolean" truevalue="booltrue" falsevalue="boolfalse" checked="false" label="Whether to use GPU?" />
+        <section name="options" title="Advanced Options" expanded="False">
+            <param argument="diameter" type="float" optional="true" label="Cell or nuclei diameter in pixels" help="Leave blank for automated estimation."/>
+            <param name="resample" type="boolean" truevalue="booltrue" falsevalue="boolfalse" checked="true" label="Run dynamics on the resampled image?"
+                help="Interpolated flows at the true image size. This option will create smoother ROIs when the cells are large but will be slower in case"/>
+            <param argument="flow_threshold" type="float" value="0.4" label="Flow error threshold (all cells with errors below threshold are kept) (not used for 3D)"/>
+            <param argument="cellprob_threshold" type="float" value="0.0" label="Cell probability threshold (all pixels with prob above threshold kept for masks)"/>
+            <param argument="niter" type="integer" value="0" label="Number of iterations" 
```

Comment:
add min/max ? also to other params?

Reply:
It is hard to find min/max for all the params. Tried to add as much as I could.